R/service-account.R

```diff
@@ -7,15 +7,24 @@
 #' @return A `httr::TokenServiceAccount` or `NULL`.
 #' @export
 credentials_service_account <- function(scopes, path = "", ...) {
+  "!DEBUG trying credentials_service account"
   if (!endsWith(path, ".json")) {
     stop("Path must end in .json")
   }
   info <- jsonlite::fromJSON(path)
-  token <- httr::TokenServiceAccount$new(NULL, info, list(scope = scopes))
+  token <- httr::TokenServiceAccount$new(
+    endpoint = NULL,
+    secrets = info,
+    ## TODO(jennybc) Is is really true I can't add the "email" scope here?
+    params = list(scope = scopes)
+  )
   if (is.null(token$credentials$access_token) ||
       !nzchar(token$credentials$access_token)) {
     NULL
   } else {
+    ## TODO(jennybc) this needs to be baked into the sub-classed service token
+    ## object, once such exists
+    message("email: ", info[["client_email"]])
     token
```

Comment:
I'm actually not sure the email needs to be baked in here, because service tokens aren't cached like OAuth tokens. Is there a reason to bother?

Reply:
My only strong feeling is that there should be a uniform way to see the email attached to any `Gargle2.0` token. The emails are long enough that one would never *type* one, but you'd definitely *confirm* one matches what you're seeing in the Cloud Console.